docs/howto/use_modules_and_layers.md

```diff
@@ -0,0 +1,270 @@
+---
+id: use_layers
+title:  "Use modules and layers"
+---
+
+# Unleash ZIO environment with `ZLayer`
+
+`ZIO` is designed around 3 parameters, `R, E, A`. `R` represents the _requirement_ for the effect to run, meaning we need to fulfill
+the requirement in order to make the effect _runnable_. We will dedicate this post to explore what we can do with `R`, as it plays a crucial role in `ZIO`.
+
+## A Simple case for `ZIO` environment
+Let's build a simple program for user management, that must get a user (if they exists), and create a user. 
+To access the DB we need a `DBConnection`, and each step in our program represents this through the environment type. We can then combine the two (small) steps through `flatMap` or more conveniently through a `for` comprehension.
+
+The result is a program that, in turn, depends on the `DBConnection`.
+
+
+```scala mdoc:invisible
+import zio.ZIO          
+import zio.IO
+import zio.UIO    
+import zio.Has
+import zio.ZEnv
+import zio.ZLayer
+import zio.clock.Clock
+import zio.console.Console
+import zio.random.Random
+
+trait DBError                  
+trait Product
+trait ProductId
+trait DBConnection
+case class UserId(value: Long)
+``` 
+
+```scala mdoc:silent
+case class User(id: UserId, name: String)
+```
+
+```scala mdoc:silent
+def getUser(userId: UserId): ZIO[DBConnection, Nothing, Option[User]] = UIO(???)
+def createUser(user: User): ZIO[DBConnection, Nothing, Unit] = UIO(???)
+
+val user: User = User(UserId(1234), "Chet")
+val created: ZIO[DBConnection, Nothing, Boolean] = for {
+  maybeUser <- getUser(user.id)
+  res       <- maybeUser.fold(createUser(user).as(true))( _ => ZIO.succeed(false))
+} yield res
+```
+
+To run the program we must supply a `DBConnection` through `provide`, before feeding it to ZIO runtime.
+
+```scala
```

Comment:
Can we use `mdoc` here. Also should be `Boolean` instead of `boolean` in type signature.

Reply:
It is difficult to use `mdoc` with `???`. I tested this snipped separately and it compiles